listenbrainz/webserver/static/js/src/recommendations/Recommendations.test.tsx

```diff
@@ -101,6 +355,7 @@ describe("handleClickNext", () => {
 
     await instance.handleClickNext();
 
+    expect(wrapper.state("loading")).toBeFalsy();
```

Comment:
This fails which is mysterious to me because `afterRecommendationsDisplay` does sets the state=`loading` to false
Can you have a look?

Reply:
So on line 349 we're overwriting the  `afterRecommendationsDisplay` method entirely, so `setState({loading:false})` is never called.

Instead, we want to spy on the existing method without replacing it, like this:
```js
const afterRecommendationsDisplaySpy = jest.spyOn(
      instance,
      "afterRecommendationsDisplay"
    );
[…]
    expect(afterRecommendationsDisplaySpy).toHaveBeenCalledTimes(1);
```